dome/install_osx_src.md

```diff
@@ -0,0 +1,296 @@
+
+## Building the Ignition Libraries in MacOS Catalina (10.15)
+
+If you want to compile Ignition Libraries in MacOS Catalina (10.15) you will need to apply some patches in your filesystem:
+
+ - `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Ruby.framework/Headers/ruby/ruby/intern.h`
+
+Create a file called `intern.patch` with the following content:
+
+```patch
+--- intern.h    2019-12-16 18:17:08.000000000 +0100
++++ /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Ruby.framework/Headers/ruby/ruby/intern.h
+@@ -14,6 +14,10 @@
+ #ifndef RUBY_INTERN_H
+ #define RUBY_INTERN_H 1
+
++#if __cplusplus > 199711L
++#define register      // Deprecated in C++11.
++#endif  // #if __cplusplus > 199711L
++
+ #if defined(__cplusplus)
+ extern "C" {
+ #if 0
+```
+
+Now we can appply the patch:
+
+```{.sh}
+sudo patch -p0 < intern.patch
+```
+
+ - `Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Ruby.framework/Headers/ruby/ruby/config.h`
+
+Create a file called `config.patch` with the following content:
+
+```
+--- config.h    2019-12-16 18:19:13.000000000 +0100
++++ Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Ruby.framework/Headers/ruby/ruby/config.h
+@@ -410,6 +410,6 @@
+ #define RUBY_PLATFORM_CPU "x86_64"
+ #endif /* defined __x86_64__ &&! defined RUBY_PLATFORM_CPU */
+ #define RUBY_PLATFORM_OS "darwin19"
+-#define RUBY_ARCH"universal-"RUBY_PLATFORM_OS
+-#define RUBY_PLATFORM"universal."RUBY_PLATFORM_CPU"-"RUBY_PLATFORM_OS
++#define RUBY_ARCH "universal-" RUBY_PLATFORM_OS
++#define RUBY_PLATFORM "universal." RUBY_PLATFORM_CPU "-" RUBY_PLATFORM_OS
+ #endif /* INCLUDE_RUBY_CONFIG_H */
+```
+
+Now we can appply the patch:
+
+```{.sh}
+sudo patch -p0 < config.patch
+```
+
+## Building the Ignition Libraries
+
+Once the compiler and all the sources are in place it is time to compile them.
+Start the procedure by changing into the workspace and listing the packages
+recognized by `colcon`:
+
+```bash
+cd ~/workspace/
+colcon graph
+```
+
+`colcon graph` should list the Ignition libraries with an
+[interdependency diagram](https://colcon.readthedocs.io/en/released/reference/verb/graph.html#example-output).
+If that is the case, then you are ready
+to build the whole set of libraries:
+
+```bash
+colcon build --merge-install
+```
+
+To speed up the build process, you could also disable tests by using
+
+```bash
+colcon build --cmake-args -DBUILD_TESTING=OFF --merge-install
+```
+
+To build a specific package with all its dependent packages:
+
+```bash
+colcon build --merge-install --packages-up-to PACKAGE_NAME
+```
+
+To build a single package:
+
+```bash
+colcon build --packages-select PACKAGE_NAME
```

Comment:
```suggestion
colcon build --merge-install --packages-select PACKAGE_NAME
```

I found that trying to run a single package after using `colcon build --merge-install` will fail unless you install the single package with `--merge-install` as well.